app/controllers/oidc/concerns/trusted_publisher_controller.rb

```diff
@@ -0,0 +1,61 @@
+module OIDC::Concerns::TrustedPublisherController
+  extend ActiveSupport::Concern
+
+  included do
+    include SessionVerifiable
+    verify_session_before
+
+    before_action :create_params, only: %i[create]
+    before_action :set_trusted_publisher_type, only: %i[create]
+    before_action :set_page, only: :index
+  end
+
+  def create_params
+    @create_params ||=
+      params
+        .permit(
+          create_params_key => [
+            :trusted_publisher_type,
+            {
+              trusted_publisher_attributes: %i[
+                repository_owner repository_name workflow_filename environment
+              ]
+            },
+            *extra_create_params
```

Comment:
Instead of composing this using inheritance and overrides, you could just accept a little duplication in the controllers and make it very readable and standard. A method for the constant attrs here could splat `*trusted_pub_permitted_params` into the rest of the permit if you want.